lib/ui/highlight-manager.js

```diff
@@ -155,27 +161,25 @@ HighlightManager.prototype = {
     //
     // Our workaround relies on using requestAnimationFrame to cut in line,
     // interleaving our highlight stealing between the setTimeouts.
+    //
     // Specifically, each time stealHighlight is called (in response to a DOM
     // mutation fired when rows are inserted into the list), we ask the browser
     // to steal the highlight three times: immediately, just before the next
     // frame is painted (the outer rAF), and just before the frame after that
     // (the inner rAF).
     //
-    // This is a weird hack, but it works fairly well; because setTimeout makes
+    // This is a weird hack, but it works fairly well. Because setTimeout makes
     // no guarantees about precisely when a callback will be executed, and
     // because the row insertion timeouts are called closely together, ours is
     // a pretty good, though nondeterministic, solution.
     if (!this.recommendation.el || this.recommendation.el.collapsed) {
       return;
     }
-    this.recommendation.isHighlighted = true;
-    this.popup.el.selectedIndex = -1;
+    this.clearHighlight();
     this.win.requestAnimationFrame(() => {
-      this.recommendation.isHighlighted = true;
-      this.popup.el.selectedIndex = -1;
+      this.clearHighlight();
       this.win.requestAnimationFrame(() => {
-        this.recommendation.isHighlighted = true;
-        this.popup.el.selectedIndex = -1;
+        this.clearHighlight();
       });
```

Comment:
This is still the biggest
![xait3izg](https://cloud.githubusercontent.com/assets/23885/14058151/5b4edcf8-f2dd-11e5-91a5-aa233d01b645.jpg)